third_party/skiplist/SkipList.cpp

```diff
@@ -0,0 +1,90 @@
+//
+//  SkipList.cpp
+//  SkipList
+//
+//  Created by Paul Ross on 19/12/2015.
+//  Copyright (c) 2017 Paul Ross. All rights reserved.
+//
+
+#include <cstdlib>
+#ifdef SKIPLIST_THREAD_SUPPORT
+#include <mutex>
+#endif
+#include <string>
+
+#include "SkipList.h"
+
+namespace duckdb_skiplistlib {
+namespace skip_list {
+
+/** Tosses a virtual coin, returns true if 'heads'.
+ *
+ * No heads, ever:
+ * @code
+ * return false;
+ * @endcode
+ *
+ * 6.25% heads:
+ * @code
+ * return rand() < RAND_MAX / 16;
+ * @endcode
+ *
+ * 12.5% heads:
+ * @code
+ * return rand() < RAND_MAX / 8;
+ * @endcode
+ *
+ * 25% heads:
+ * @code
+ * return rand() < RAND_MAX / 4;
+ * @endcode
+ *
+ * Fair coin:
+ * @code
+ * return rand() < RAND_MAX / 2;
+ * @endcode
+ *
+ * 75% heads:
+ * @code
+ * return rand() < RAND_MAX - RAND_MAX / 4;
+ * @endcode
+ *
+ * 87.5% heads:
+ * @code
+ * return rand() < RAND_MAX - RAND_MAX / 8;
+ * @endcode
+ *
+ * 93.75% heads:
+ * @code
+ * @return rand() < RAND_MAX - RAND_MAX / 16;
+ * @endcode
+ */
+bool tossCoin() {
+    return rand() < RAND_MAX / 2;
+}
+
+void seedRand(unsigned seed) {
+    srand(seed);
+}
```

Comment:
This seems a strange choice, since it's somehow wasteful to compute N bits of randomness to use only 1 of them. But I don't think this is ever relevant for real.
Slightly more problematic, this messes with the state of the random generator, that I would probably avoid (`seedRand` is never actually called luckily), plus I think this might forces synchronisation cross threads, while here we are fine with thread local not-really random sources.

I would have gone with:
```
std::independent_bits_engine<std::default_random_engine, /*bits*/ 1, int> tossCoin;
if (tossCoin())  {
   // yey!
} else {
    // noo
}
```

(I went a bit in-depth on a not really relevant part)

Reply:
Interesting observation. I suppose since pandas is single-threaded this was not an issue for them. I suppose I could stuff in a thread-safe prng (I think we have one?)

Reply:
Reverting this because it trashes performance.